### PR TITLE
update offset_of! docs to reflect the stabilization of nesting

### PR DESCRIPTION
**Pull request by lolbinarycat**
this seems to have been missed.

### CODE REVIEW

**Comment by workingjubilee** [comment]
There seems to be a separate `offset_of_enum` feature?

**Comment by lolbinarycat** [comment]
fixed.  technically separate issues, but they're both trivial, so might as well fix both at the same time.

File: library/core/src/mem/mod.rs [hunk 1]
```diff
@@ -1254,12 +1254,10 @@ impl<T> SizedTypeProperties for T {}
 ///
 /// Nested field accesses may be used, but not array indexes.
 ///
-/// Enum variants may be traversed as if they were fields. Variants themselves do
+/// If the nightly-only feature `offset_of_enum` is enabled,
+/// variants may be traversed as if they were fields. Variants themselves do
 /// not have an offset.
```
Comment (workingjubilee) [inline library/core/src/mem/mod.rs:1259]: one last nit: while we're here, can we semantically linebreak this? https://sembr.org/
```suggestion
/// variants may be traversed as if they were fields.
/// Variants themselves do not have an offset.
```

**Comment by workingjubilee** [comment]
I just wanted to be sure I was approving the correct thing, since it seemed like it could confuse people into thinking no functionality was feature-gated at all.

**Comment by workingjubilee** [comment]
? why didn't I...

@bors r+ rollup

**Pull request merged**